lib/Parse/Lexer.cpp

```diff
@@ -2039,25 +2039,17 @@ bool Lexer::tryLexRegexLiteral(const char *TokStart) {
 
   // Ask the Swift library to try and lex a regex literal.
   // - Ptr will not be advanced if this is not for a regex literal.
-  // - ErrStr will be set if there is any error to emit.
   // - CompletelyErroneous will be set if there was an error that cannot be
   //   recovered from.
   auto *Ptr = TokStart;
-  const char *ErrStr = nullptr;
-  bool CompletelyErroneous = regexLiteralLexingFn(&Ptr, BufferEnd, &ErrStr);
+  bool CompletelyErroneous = regexLiteralLexingFn(
+      &Ptr, BufferEnd, getBridgedOptionalDiagnosticEngine(getTokenDiags()));
 
   // If we didn't make any lexing progress, this isn't a regex literal and we
   // should fallback to lexing as something else.
   if (Ptr == TokStart)
     return false;
 
-  if (ErrStr) {
-    if (!MustBeRegex)
-      return false;
```

Comment:
@hamishknight This erasure didn't fail the current test cases, do you come up with a test case this should be in effect?

Reply:
Huh, I thought we had this test case in `forward-slash-regex.swift` but I guess not. This check allows us to bail if the closing delimiter is missing in e.g:

```swift
func foo(_ x: (Int, Int) -> Int, _ y: Int) {}
foo(/, 0)
```

Reply:
Restored the behavior and added this test case to `forward-slash-regex.swift`